internal/clients/aws.go

```diff
@@ -230,18 +99,57 @@ func (m *metaOnlyPrimary) Meta() any {
 	return m.meta
 }
 
-func configureNoForkAWSClient(ctx context.Context, ps *terraform.Setup, config *SetupConfig) error { //nolint:gocyclo
-	p := *config.TerraformProvider
-	diag := p.Configure(context.WithoutCancel(ctx), &tfsdk.ResourceConfig{
-		Config: ps.Configuration,
+// configureNoForkAWSClient populates the supplied *terraform.Setup with
+// Terraform Plugin SDK style AWS client (Meta) and Terraform Plugin Framework
+// style FrameworkProvider
+func configureNoForkAWSClient(ctx context.Context, ps *terraform.Setup, config *SetupConfig, awsCfg *aws.Config, creds aws.Credentials, pc *v1beta1.ProviderConfig) error { //nolint:gocyclo
+	tfAwsConnsCfg := xpprovider.AWSConfig{
+		AccessKey:                     creds.AccessKeyID,
+		EC2MetadataServiceEnableState: imds.ClientDefaultEnableState,
+		Endpoints:                     map[string]string{},
+		Region:                        awsCfg.Region,
+		S3UsePathStyle:                pc.Spec.S3UsePathStyle,
+		SecretKey:                     creds.SecretAccessKey,
+		SkipCredsValidation:           true, // disabled to prevent extra AWS STS call
+		SkipRegionValidation:          pc.Spec.SkipRegionValidation,
+		SkipRequestingAccountId:       true, // disabled to prevent extra AWS STS call
+		Token:                         creds.SessionToken,
+	}
+
+	if pc.Spec.SkipMetadataApiCheck {
+		tfAwsConnsCfg.EC2MetadataServiceEnableState = imds.ClientEnabled
+	}
+
+	if pc.Spec.Endpoint != nil {
+		if pc.Spec.Endpoint.URL.Static != nil {
+			if len(pc.Spec.Endpoint.Services) > 0 && *pc.Spec.Endpoint.URL.Static == "" {
+				return errors.New("endpoint.url.static cannot be empty")
+			} else {
+				for _, service := range pc.Spec.Endpoint.Services {
+					tfAwsConnsCfg.Endpoints[service] = aws.ToString(pc.Spec.Endpoint.URL.Static)
+				}
+			}
+		}
+	}
+
+	// only used for retrieving the ServicePackages from the singleton provider instance
+	p := config.TerraformProvider.Meta()
+	tfAwsConnsClient, diags := tfAwsConnsCfg.GetClient(ctx, &xpprovider.AWSClient{
+		// #nosec G103
+		ServicePackages: (*xpprovider.AWSClient)(unsafe.Pointer(reflect.ValueOf(p).Pointer())).ServicePackages,
```

Comment:
If I remember correctly, this service packages concept played a role in properly handling the resource tags. Have you also tested if resource tags are still being handled properly, e.g., are the default tags and tag updates still working as expected?

Reply:
Yes, double-checked this and the tags are shown up in console and changes are reflected correctly.